crates/uv-python/src/discovery.rs

```diff
@@ -1490,6 +1473,106 @@ impl PythonRequest {
         Self::ExecutableName(value.to_string())
     }
 
+    /// Try to parse a tool executable as a Python version, e.g. `uvx [executable]`.
+    ///
+    /// The `PythonRequest::parse` constructor above is intended for the `--python` flag, where the
+    /// value is unambiguously a Python version. This alternate constructor is intended for `uvx`,
+    /// where the executable might be a Python version or a package name. There are several
+    /// differences in behavior:
+    /// - This only supports long names, including e.g. `pypy39` but **not** `pp39` or `39`.
+    /// - On Windows only, this allows `pythonw` as an alias for `python`.
+    /// - This allows `python` by itself (and on Windows, `pythonw`) as an alias for `default`.
+    ///
+    /// This can only return `Err` if `@` is used, see try_parse_tool_executable. Otherwise, if no
+    /// match is found, it returns `Ok(None)`.
+    pub fn try_parse_tool_executable(value: &str) -> Result<Option<PythonRequest>, Error> {
```

Comment:
Maybe `try_from_executable_name`?

Reply:
I'm also just about to push a commit that adds back `uvx --from python[...]` support, and this function gets used in that case too, where its argument isn't really an "executable" or a "command". I'm going to go with `try_from_tool_name`. Let me know what you think.